cmd/sonictool/db/dbutils.go

```diff
@@ -61,15 +63,23 @@ func MakeDbProducer(chaindataDir string, cacheRatio cachescale.Func) (kvdb.FullD
 	})
 }
 
-func MakeGossipDb(dbs kvdb.FullDBProducer, dataDir string, validatorMode bool, cacheRatio cachescale.Func) (*gossip.Store, error) {
+func MakeGossipDb(ctx *cli.Context, dbs kvdb.FullDBProducer, dataDir string, validatorMode bool, cacheRatio cachescale.Func) (*gossip.Store, error) {
```

Comment:
This is odd, but I see that it is some kind of pattern in this part of the code:
Passing around the cli.context is like sharing the command line arguments over. Code consuming some flag will retrieve the value from the command line arguments and validate it in situ. This may be too late to provide meaningful error reports to the user to correct a faulty command line entry.